wlauto/utils/power.py

```diff
@@ -515,6 +515,39 @@ def write(self, filepath):
                                            for s in stats])
 
 
+class CpuUtilisationTimeline(object):
+
+    def __init__(self, filepath, core_names):
+        self.filepath = filepath
+        self._wfh = open(filepath, 'w')
+        self.writer = csv.writer(self._wfh)
+        if core_names:
+            headers = ['ts'] + ['{} CPU{}'.format(c, i)
+                                for i, c in enumerate(core_names)]
+            self.writer.writerow(headers)
+        self._rows = []
+        self._max = 1
+
+    def update(self, timestamp, core_states):  # NOQA
+        row = [timestamp]
+        for idle_state, frequency in core_states:
+            if idle_state == -1:
+                if frequency == UNKNOWN_FREQUENCY:
+                    frequency = 0
+            elif idle_state is None:
+                frequency = 0
+            else:
+                frequency = 0
+            row.append(frequency)
+            self._max = max(self._max, frequency)
```

Comment:
The max frequency can vary for different clusters meaning littles would never show 100% utilisation even though they are at 100%. Also if the system is running idle this would report higher utilisation than is actually occuring. You should get the max frequency out of cpufreq (which is accessible via the WA device). This should also error if cpufreq is not available on a device.